modules/api/src/main/java/org/apache/ignite/table/manager/IgniteTables.java

```diff
@@ -58,41 +59,12 @@
      */
     CompletableFuture<Table> createTableAsync(String name, Consumer<TableChange> tableInitChange);
 
-    /**
-     * Creates a new table with the given {@code name} or returns an existing one with the same {@code name}.
-     *
-     * <p>Note: the configuration of the existed table will NOT be validated against the given {@code tableInitChange}.
-     *
-     * @param name            Table name.
-     * @param tableInitChange Table changer.
-     * @return Existing or newly created table.
-     * @throws IgniteException If an unspecified platform exception has happened internally. Is thrown when:
-     *                         <ul>
-     *                             <li>the node is stopping.</li>
-     *                         </ul>
-     */
-    Table createTableIfNotExists(String name, Consumer<TableChange> tableInitChange);
-
-    /**
-     * Creates a new table with the given {@code name} or returns an existing one with the same {@code name}.
-     *
-     * <p>Note: the configuration of the existed table will NOT be validated against the given {@code tableInitChange}.
-     *
-     * @param name            Table name.
-     * @param tableInitChange Table changer.
-     * @return Future representing pending completion of the operation.
-     * @throws IgniteException If an unspecified platform exception has happened internally. Is thrown when:
-     *                         <ul>
-     *                             <li>the node is stopping.</li>
-     *                         </ul>
-     */
-    CompletableFuture<Table> createTableIfNotExistsAsync(String name, Consumer<TableChange> tableInitChange);
-
     /**
      * Alter a cluster table.
      *
      * @param name        Table name.
      * @param tableChange Table changer.
+     * @throws TableNotExistsException If a table with the name does not exist.
```

Comment:
Seems that in case of async methods TableNotExistsException isn't actually thrown, but used in order to complete the future, is that correct?

Reply:
This is a javadoc for synchronous method.

Reply:
Yep, u a right, my fault.